Tak.py

```diff
@@ -62,8 +64,18 @@ def terminal_test(self):
                     if len(space) > 0:      #There's at least 1 piece in the current location
                         if space[len(space)-1].type != pieceType.WALL: #Piece can be part of road
                             if self.findRoads(space[len(space)-1]):
-                                print(f"Road Found! Player: {space[len(space)-1].color}")
-                                return True
+                                playerRoads.add(space[len(space)-1].color)
+        if len(playerRoads):
+            #player who made the winning move gets the win, regardless of whether not enemy also had a road
+            if playerRoads.intersection({GameState.to_move}):
+                if len(playerRoads)==2:
```

Comment:
What if we have a scenario with more than 2 roads? Just placing a piece that couldn't happen, but if a stack was moved I think it could.

Suppose the top row is empty. The rightmost column (except for the top space) are white. Every other space not yet mentioned is black. Then if we have a stack in the top left (instead of being empty) that has a white on top and 4 blacks below. White could moved the stack over to have the board end up in a state where the rightmost column is white, and all else is black, with 5 rows total.

I think just having a boolean value set to true if the current player made a road would work, then if there are multiple roads and the boolean is true the player wins, else the other one does.

Reply:
Or just move through all of the elements in `playerRoads` and see if there are 2 colors there or, if it's easier, if the current player's color is there. A lamda function like that should be pretty readable.